### PR TITLE
fix: Update the Monitor, proposer and disputer bots to work with OOv2 contracts

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

This PR updates the monitor, disputer and proposer bots to work with the OOv2.

**Summary**

Bot and test updated.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [X]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/monitors/src/OptimisticOracleContractMonitor.js [hunk 1]
```diff
@@ -338,7 +338,19 @@ class OptimisticOracleContractMonitor {
   }
 
   _generateUILink(transactionHash, chainId) {
-    return `<${this.optimisticOracleUIBaseUrl}/request?transactionHash=${transactionHash}&chainId=${chainId}|View in UI>`;
```
Comment (chrismaree) [inline packages/monitors/src/OptimisticOracleContractMonitor.js:341]: note this change: the OO UI now needs different URLS based on the OO type. this switch case supports this.

File: packages/financial-templates-lib/test/clients/OptimisticOracleClient.js [hunk 2]
```diff
@@ -104,34 +105,35 @@ describe("OptimisticOracleClient.js", function () {
     // Set a non-0 final fee for the collateral currency.
     await store.methods.setFinalFee(collateral.options.address, { rawValue: finalFee }).send({ from: accounts[0] });
 
-    optimisticOracle = await OptimisticOracle.new(liveness, finder.options.address, timer.options.address).send({
```
Comment (chrismaree) [inline packages/financial-templates-lib/test/clients/OptimisticOracleClient.js:107]: I re-structured these tests to be a bit cleaner by correctly using the before each.

File: packages/financial-templates-lib/test/clients/OptimisticOracleClient.js [hunk 3]
```diff
@@ -685,6 +705,343 @@ describe("OptimisticOracleClient.js", function () {
       await advanceBlockAndSetTime(web3, currentTime + 1);
       await advanceBlockAndSetTime(web3, currentTime + 2);
 
+      await clientShortLookback.update();
+      result = clientShortLookback.getUnproposedPriceRequests();
+      objectsInArrayInclude(result, []);
+    });
+  });
+  describe("OptimisticOracleV2 contract", function () {
```
Comment (chrismaree) [inline packages/financial-templates-lib/test/clients/OptimisticOracleClient.js:713]: for all intents and purposes this test (and those in the other implementations) simply copy/paste the existing tests and update them to show they still work with the v2 contrcts.

File: packages/financial-templates-lib/test/clients/OptimisticOracleEventClient.js [hunk 4]
```diff
@@ -67,6 +70,7 @@ describe("OptimisticOracleEventClient.js", function () {
 
   const pushPrice = async (price) => {
     const [lastQuery] = (await mockOracle.methods.getPendingQueries().call()).slice(-1);
+    console.log("lastQuery", lastQuery);
```
Comment (md0x) [inline packages/financial-templates-lib/test/clients/OptimisticOracleEventClient.js:73]:
```suggestion
```

**Review by md0x — APPROVED** [review]
LGTM

**Review by nicholaspai — APPROVED** [review]
This looks good to me! It's mostly test changes. Do you have any places in particular you want me to review?

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**